mat64/list_test.go

```diff
@@ -0,0 +1,706 @@
+						panic("makeRandOf returns wrong size" + fmt.Sprintf("%T\n", bMat))
+						continue
+					}
+				}
+
+				// Compute the true answer if the sizes are legal.
+				legal := legalSize(test.ar, test.ac, test.br, test.bc)
+				var truth Dense
+				if legal {
+					var aDense, bDense Dense
+					aDense.Clone(a)
+					bDense.Clone(b)
+					denseComparison(&truth, &aDense, &bDense)
+				}
+				// Call the method and check the result with all transpose variations.
+				// Do this by explicitly create a transpose, and then calling T().
+				for _, aTrans := range []bool{true, false} {
+					for _, bTrans := range []bool{true, false} {
+						aCopy := makeCopyOf(a)
+						bCopy := makeCopyOf(b)
+						// Set the proper function call
+						var f func(receiver, a, b Matrix)
+						if aTrans {
+							if bTrans {
+								f = func(receiver, a, b Matrix) {
+									aT := makeTransposeCopyOf(a)
+									r, c := a.Dims()
+									tr, tc := aT.Dims()
+									if r != tc || c != tr {
+										panic("makeTransposeCopyOf returns wrong size" + fmt.Sprintf("%T", a))
+									}
+									bT := makeTransposeCopyOf(b)
+									r, c = b.Dims()
+									tr, tc = bT.Dims()
+									if r != tc || c != tr {
+										panic("makeTransposeCopyOf returns wrong size" + fmt.Sprintf("%T", b))
+									}
+									method(receiver, aT.T(), bT.T())
+								}
+							} else {
+								f = func(receiver, a, b Matrix) {
+									aT := makeTransposeCopyOf(a)
+									method(receiver, aT.T(), b)
+								}
+							}
+						} else {
+							if bTrans {
+								f = func(receiver, a, b Matrix) {
+									bT := makeTransposeCopyOf(b)
+									method(receiver, a, bT.T())
+								}
+							} else {
+								f = func(receiver, a, b Matrix) {
+									method(receiver, a, b)
+								}
+							}
+						}
+						// Test the method for a zero-value of the receiver.
+						errStr := fmt.Sprintf("%T(%T, %T), sizes: %#v, atrans %v, btrans %v", receiver, a, b, test, aTrans, bTrans)
+						zero := makeRandOf(receiver, 0, 0)
+						p, err := panics(func() { f(zero, a, b) })
+						if !legal && !p {
+							c.Errorf("Did not panic with illegal size: " + errStr)
+							continue
+						}
+						if legal && p {
+							c.Errorf("Panicked with legal size: " + errStr + err)
+							continue
+						}
+						if !equalApprox(zero, &truth, 1e-14) {
+							c.Errorf("Answer mismatch with zero receiver:" + errStr)
+							continue
+						}
+						if !equal(a, aCopy) {
+							c.Errorf("First input argument changed in call: " + errStr)
+						}
+						if !equal(b, bCopy) {
+							c.Errorf("Second input argument changed in call: " + errStr)
+						}
+
+						if !legal {
+							continue
+						}
+
+						// Test the method with a non-zero-value of the receiver.
+						// The receiver has been overwritten in place so we can
+						// take its new size.
+						rr, rc := zero.Dims()
+						nonzero := makeRandOf(receiver, rr, rc)
+						p, err = panics(func() { f(nonzero, a, b) })
```

Comment:
Done.

What's the convention with moving this into the if statement? 
````
if panicked, _ := panic(); panicked{
}
````
On the one hand, panicked is ephemeral, but on the other hand nonZero is being meaningfully modified in the function call.